packages/aws-cdk/lib/init-templates/v2/app/go/cdk.template.json

```diff
@@ -1,3 +1,6 @@
 {
-  "app": "go mod download && go run %name%.go"
+  "app": "go mod download && go run %name%.go",
+  "watch": {
+    "include": "**/*.go"
```

Comment:
I actually don't know if the Go template includes tests by default, but worth checking.